lib/TemplatedPathPlugin.js

```diff
@@ -338,7 +340,29 @@ class TemplatedPathPlugin {
 	 */
 	apply(compiler) {
 		compiler.hooks.compilation.tap(plugin, compilation => {
-			compilation.hooks.assetPath.tap(plugin, replacePathVariables);
+			compilation.hooks.assetPath.tap(
+				plugin,
+				(pathTemplate, data, assetInfo) => {
+					const path = replacePathVariables(pathTemplate, data, assetInfo);
```

Comment:
this could be not a path..
e.g. in case of source maps (strings like //# sourceMappingURL=[url] )